report2/part2/reports.py

```diff
@@ -0,0 +1,72 @@
+import math
+# Report functions
+
+def get_most_played(file_name):
+    game = open("game_stat.txt", "r")
+    game_inv = game.readlines()
+    game.close()
+    maximum = max(float(line.split("\t")[1]) for line in game_inv)
+    for line in game_inv:
+        if float(line.split("\t")[1]) == maximum:
+            return line.split("\t")[0]
+
+def sum_sold(file_name):
+    with open(file_name) as game_inv:
+        return sum(float(x[1]) for x in [line.split("\t") for line in game_inv])
+
+def get_selling_avg(file_name):
+    game = open("game_stat.txt", "r")
+    game_inv = game.readlines()
+    game.close()
+    count = sum(1 for line in game_inv)
+    return sum(float(line.split("\t")[1]) for line in game_inv) / count
+
+def count_longest_title(file_name):
+    with open(file_name) as game_inv:
+        return len(max((x[0] for x in [line.split("\t") for line in game_inv]), key = len))
+
+def get_date_avg(file_name):
+    game = open("game_stat.txt", "r")
+    game_inv = game.readlines()
+    game.close()
+    count = sum(1 for line in game_inv)
+    return math.ceil(sum(int(line.split("\t")[2]) for line in game_inv) / count)
+
+def get_game(file_name, title):
+    game = open("game_stat.txt", "r")
+    game_inv = game.readlines()
+    game.close()
+    game_rows = [x for x in [line.strip("\n").split("\t") for line in game_inv] if x[0] == title]
```

Comment:
hasznald a [next(...)](http://stackoverflow.com/questions/9868653/find-first-list-item-that-matches-criteria)-et:
```
game_raw = next(line for line in game_inv if line.strip("\n").split("\t")[0] == title)
game = [game_raw[0], float(game_raw[1]), ...]
```